utils/db.ts

```diff
@@ -0,0 +1,146 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+export const kv = await Deno.openKv();
+
+const versionstamp = null;
+
+export interface InitItemValue {
+  userId: string;
+  title: string;
+  url: string;
+}
+
+export async function createItem(initItem: InitItemValue) {
+  const itemId = crypto.randomUUID();
+
+  const itemKey = ["items", crypto.randomUUID()];
+  const itemsByUserKey = ["items_by_user", initItem.userId, itemId];
+
+  const item: ItemValue = { ...initItem, score: 0, createdAt: new Date() };
+  const res = await kv.atomic()
+    .check({ key: itemKey, versionstamp })
+    .check({ key: itemsByUserKey, versionstamp })
+    .set(itemKey, item)
+    .set(itemsByUserKey, item)
+    .commit();
+
+  if (res === null) {
+    throw new TypeError("Item with ID or for user already exists");
+  }
```

Comment:
Instead of erroring, why not run this in a loop until you find a UUID that isn't in use?

Reply:
Good catch. Done.